### PR TITLE
Migrate UI tests to JUnit 5

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UISuite.scala [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 
   registerFixtureTransformer(_.withAfterIntelliJStartup((_, intelliJ) => intelliJ.ide.configure()))
 
   override protected def baseFixture: IntelliJFixture = fixtureFromConfig("ideprobe.conf")
 
+  // Don't call it `beforeAll()`, such method already exists in `RunningIntelliJPerSuiteBase`
+  @BeforeAll override final def setup(): Unit = super.setup()
+
+  // Don't call it `afterAll()`, such method already exists in `RunningIntelliJPerSuiteBase`
+  @AfterAll override final def teardown(): Unit = super.teardown()
```
Comment (PawelLipski) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UISuite.scala:16]: No need to create junit5-driver in ide-probe... that' all what's needed as a replacement ;p

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 2]
```diff
@@ -15,19 +14,45 @@ import java.nio.file.attribute.FileTime
 import scala.language.postfixOps
 import scala.sys.process._
 
-object UITestSuite extends UISuite {}
+class ScreenshotAndThreadDumpExtension extends TestWatcher {
 
-@RunWith(classOf[JUnit4])
+  override def testSuccessful(context: ExtensionContext): Unit = {
```
Comment (PawelLipski) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:19]: To be removed, lemme check if this `TestWatcher` really works

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 3]
```diff
@@ -15,19 +14,45 @@ import java.nio.file.attribute.FileTime
...
+    Files.createDirectories(Paths.get(artifactDirectory))
+    val file: File = new File(artifactDirectory + "/thread_dump_" + pid + ".txt")
+    val pw = new PrintWriter(file)
+    pw.write(threadStackTrace)
+    pw.close()
+  }
+}
+
+object UITestSuite extends UISuite {
+  // We need to declare a companion object so that JUnit sees @BeforeAll/@AfterAll static methods.
```
Comment (PawelLipski) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:42]: Was also true under JUnit 4 as well for `@BeforeClass`/`@AfterClass`

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala (hunk 2 above)
Comment (PawelLipski) [reply src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala]: Yes it does

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 4]
```diff
@@ -127,7 +127,9 @@ class UITestSuite extends TestGitRepository(SETUP_WITH_SINGLE_REMOTE) {
     )
     branchAndCommitRowsCount = project.refreshModelAndGetRowCount()
     // 5 branch rows (`call-ws` is also no longer there) + 7 commit rows
-    Assert.assertEquals(12, branchAndCommitRowsCount)
+    assertEquals(12, branchAndCommitRowsCount)
+
+    fail("lol xd")
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala]: Yes it does, https://app.circleci.com/pipelines/github/VirtusLab/git-machete-intellij-plugin/8329/workflows/d627ced7-8494-4438-b977-44f26bae5674/jobs/8755/artifacts

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/hooks/StatusBranchHookExecutor.java [hunk 5]
```diff
@@ -31,7 +31,7 @@
 
 @CustomLog
 public final class StatusBranchHookExecutor extends BaseHookExecutor {
-  private static final int EXECUTION_TIMEOUT_SECONDS = 1;
+  private static final int EXECUTION_TIMEOUT_SECONDS = 2;
```
Comment (PawelLipski) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/hooks/StatusBranchHookExecutor.java:34]: Unrel, mostly for the sake of `backend:impl` integration tests passing more reliably

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/errorreport/PlatformInfoProvider.java [hunk 6]
```diff
@@ -25,6 +25,6 @@ String getIdeApplicationName() {
   @Nullable
   String getPluginVersion() {
     IdeaPluginDescriptor pluginDescriptor = PluginManagerCore.getPlugin(PluginId.getId("com.virtuslab.git-machete"));
-    return pluginDescriptor != null ? pluginDescriptor.getVersion() : "<unknown>";
+    return pluginDescriptor != null ? pluginDescriptor.getVersion() : null;
```
Comment (PawelLipski) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/errorreport/PlatformInfoProvider.java:28]: Unrel, spotted while reviewing the release PR

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -14,8 +13,7 @@ betterStrings = "com.antkorwin:better-strings:0.5"
 checker = { module = "org.checkerframework:checker", version.ref = "checker" }
 checker-qual = { module = "org.checkerframework:checker-qual", version.ref = "checker" }
 commonsIO = "commons-io:commons-io:2.11.0"
-ideProbe-junitDriver = { module = "org.virtuslab.ideprobe:junit-driver_2.13", version.ref = "ideProbe" }
-ideProbe-robotDriver = { module = "org.virtuslab.ideprobe:robot-driver_2.13", version.ref = "ideProbe" }
+ideProbe-robotDriver = "org.virtuslab.ideprobe:robot-driver_2.13:0.51.0"
```
Comment (mkondratek) [inline gradle/libs.versions.toml:16]: wow - it's code removal btw 😮

File: src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala [hunk 8]
```diff
@@ -15,19 +14,38 @@ import java.nio.file.attribute.FileTime
 import scala.language.postfixOps
 import scala.sys.process._
 
-object UITestSuite extends UISuite {}
+class ScreenshotAndThreadDumpExtension extends TestWatcher {
```
Comment (mkondratek) [inline src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala:17]: nice

**Review by mkondratek — APPROVED** [review]
👍

**Pull request merged**